src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -295,6 +315,24 @@ void ClusterTaskManager::DispatchScheduledTasksToWorkers(
         continue;
       }
 
+      if (sched_cls_info.running_tasks.size() >= sched_cls_info.capacity) {
+        if (get_time_() > sched_cls_info.next_update_time) {
```

Comment:
Shouldn't we reset next_update_time if the running tasks drops below the capacity? Otherwise eventually naturally enough time will pass that we exceed this time immediately on hitting capacity.